wfl/cli/commands/error.py

```diff
@@ -1,6 +1,89 @@
+         " ``/comp`` for different components being counted separately. Default of [\"energy/atom\", \"virial/atom/comp\"]only used if neither ``config_properties``"\
+         " nor ``atom_properties`` is present.")
+@click.option("--atom-properties", '-ap', multiple=True, 
+    help="Multiple `Atoms.arrays` calculated properties (to be prefixed by ``ref_property_prefix``"\
+         "or ``calc_property_prefix``) to compute error For.  Properties can end with ``/comp``"\
+         "for different components being computed separately, and ``/Z`` for different atomic numbers"\
+         "to be assigned to different categories.  Default of [\"forces\"] only used if neither ``config_properties``"\
+         "nor ``atom_properties`` is present.")
+@click.option("--category-keys", '-ck', multiple=True, default=["config_type"], show_default=True,
+    help="Results will be averaged by a category, defined by a string containing the values of these"\
+         "keys in Atoms.info, in addition to overall average _ALL_ category.")
+@click.option("--weight-property", '-wp', help="Atoms.info key for weights to apply to RMSE calculation")
+@click.option("--precision", '-p', type=click.INT, default=2, help="precision for printing table")
+@click.option("--fig-name", "-f", help="Filename for figure. Do not plot if not given. ")
+@click.option("--error-type", default="RMSE", show_default=True, type=click.Choice(["RMSE", "MAE"]), 
+    help="Which error to report in legend")
+@click.option("--cmap", help="Colormap to use for plot if not default")
+@click.pass_context
+@opt.inputs
+def show_error(ctx, inputs, calc_property_prefix, ref_property_prefix, 
+          config_properties, atom_properties, category_keys,
+          weight_property, precision, fig_name, error_type, 
+          cmap):
+    """Prints error summary table"""
+    # TODO
+    # - clean up cmap
+
+    errors, diffs, parity = ref_err_calc(
+        inputs=inputs, 
+        calc_property_prefix=calc_property_prefix,
+        ref_property_prefix=ref_property_prefix,
+        config_properties=config_properties,
+        atom_properties=atom_properties,
+        category_keys=category_keys,
+        weight_property=weight_property)
+
+    df = format_errors(errors)
+    print_df(df, precision)
+
+    if fig_name:
+
+        value_error_scatter(
+            all_errors = errors, 
+            all_diffs=diffs,
+            all_parity=parity,
+            output=fig_name,
+            ref_property_prefix=ref_property_prefix,
+            calc_property_prefix=calc_property_prefix,
+            error_type = error_type,
+            cmap=cmap
+        )
+
+
+def print_df(df, precision=3):
+
+    pd.set_option('display.max_rows', None)
+    pd.set_option('display.max_columns', None)
+    pd.set_option('display.colheader_justify', 'center')
+    pd.set_option('display.precision', precision)
+    print(df)
+
+
+def format_errors(errors):
```

Comment:
Might be nice to have this and `print_df` available from python, not just command line. Would just need to move them into `fit/error.py` and call those routines from here.  It might also be nice to add an optional file argument to `print_df`, defaulting to `sys.stdout`. Or we could have a version that returns a formatted string with `io.StringIO`.

Also, `pd.set_option` is a global setting, which wouldn't be so friendly if it's in a function, not the command line, so we'd need to figure out how to get pandas to apply the options only to this table output operation.   I think you can use `DataFrame.to_string()` with some optional arguments that take the place of these options.

Reply:
Yes - I think the optimal thing is to have `format_errors()` return a string, probably generated by `df.to_string(..., max_rows=...,...)` with other optional args, and move that function to `fit/error.py`.  Then the command line can just print that string, and other people can call the function as well and print it wherever they want.

Reply:
> Might be nice to have this and print_df available from python, not just command line. 

> It might also be nice to add an optional file argument to `print_df`, defaulting to `sys.stdout`. 

sounds good. 

> `pd.set_option` is a global setting

Ah of course. I was going to fix that with a `with` scope, but I guess I can convert to correctly formatted string before printing. I would rather `format_errors()` returned a dataframe though, because then other people have options to convert it to markdown, html, latex, ... not only pre-formated string.

Reply:
Added this in [e8d082b](https://github.com/libAtoms/workflow/pull/192/commits/e8d082be11a733f1ec871dc78782bb8b2988f966)

Reply:
I'd still prefer to have a routine that returns a string.  See my comment below.